### ISSUE TITLE
[Client bug]: LargeFileUploadTask - Blazor WASM - Operation is not supported on this platform.

### ISSUE THREAD

**Issue by ivancavlek** [microsoftgraph/msgraph-sdk-dotnet#966]
**Describe the bug**
I want to upload a file from Blazor WASM to SharePoint Online with C# Client from Microsoft Graph. When I arrive at LargeFileUploadTask point in Code I get _Operation is not supported on this platform._.
I am using .NET 5.

**To Reproduce**
I basically follow [this](https://docs.microsoft.com/en-us/graph/sdks/large-file-upload?tabs=csharp) code example which I integrated into my Blazor code.

```C#
async Task<ContainerItemDto> ISharePointFileService.UploadFileAsync(
            string entityFolderName, string uidFolderName, string fileName, byte[] fileBytes)
{
    var usersStore = Country.All;//await _userService.GetUsersStore().ConfigureAwait(false);
    var storeDriveId = GetStoreDrive(usersStore);
    var entityFolderId = await GetOrCreateFolder("root", entityFolderName).ConfigureAwait(false);
    var uidFolderId = await GetOrCreateFolder(entityFolderId, uidFolderName).ConfigureAwait(false);
    fileName = GetSharePointFileStoreName(fileName);
    var uploadProperties = new DriveItemUploadableProperties
    {
        ODataType = null,
        AdditionalData = new Dictionary<string, object>
        {
            { "@microsoft.graph.conflictBehavior", "replace" }
        }
    };

    var uploadSession = await _httpClient
        .Drives[storeDriveId]
        .Items[uidFolderId]
        .ItemWithPath(fileName)
        .CreateUploadSession(uploadProperties)
        .Request()
        .PostAsync()
        .ConfigureAwait(false);

    int maxSliceSize = 320 * 1024;
    using var memoryStream = new MemoryStream(fileBytes);
    var fileUploadTask = new LargeFileUploadTask<DriveItem>(uploadSession, memoryStream, maxSliceSize); // Error!!!

    var uploadResult = await fileUploadTask.UploadAsync(/*progress*/).ConfigureAwait(false);

    return uploadResult.UploadSucceeded ?
        new ContainerItemDto
        {
            Id = uploadResult.ItemResponse.Id,
            DisplayName = fileName,
            Link = uploadResult.ItemResponse.WebUrl
        } :
        throw new ArgumentNullException(nameof(uploadResult));
}
```
This results in:
_Operation is not supported on this platform._

If I were to utilize the fourth parameter to LargeFileUploadTask:
```C#
var fileUploadTask = new LargeFileUploadTask<DriveItem>(uploadSession, memoryStream, maxSliceSize, _httpClient);
```
I receive _System.NullReferenceException: Object reference not set to an instance of an object._.

**Expected behavior**
I expect the file to be uploaded and to receive the file metadata from SharePoint Online (Id and WebUrl).

**Screenshots**
_Operation is not supported on this platform._
```C#
var fileUploadTask = new LargeFileUploadTask<DriveItem>(uploadSession, memoryStream, maxSliceSize);
```
![image](https://user-images.githubusercontent.com/16662190/115044021-62ef3a80-9ed5-11eb-8a10-7240970b86d1.png)

_System.NullReferenceException: Object reference not set to an instance of an object._.
```C#
var fileUploadTask = new LargeFileUploadTask<DriveItem>(uploadSession, memoryStream, maxSliceSize, _httpClient);
```
![image](https://user-images.githubusercontent.com/16662190/115043587-f3794b00-9ed4-11eb-8c45-81cdd70df49e.png)

**Client version**
Microsoft.Graph v3.29.0

**Desktop (please complete the following information):**
 - Windows 10
 - Edge
 - Version 89.0.774.75 (Official build) (64-bit)

**Additional context**
I am using Microsoft Graph intensively from Blazor WASM and everything works just fine.

Could you, please, give me an initial feedback if Blazor WASM is supported by LargeFileUploadTask and a workaround if it is not? 
[AB#9034](https://microsoftgraph.visualstudio.com/0985d294-5762-4bc2-a565-161ef349ca3e/_workitems/edit/9034)

**Comment by andrueastman** [comment]
@ivancavlek From your screenshots it looks like the uploadSession is not created successfully since the POST call results in a 404 response. 
Are you able to confirm that the uploadSession object is valid and has valid properties?

**Comment by ivancavlek** [comment]
Hi @andrueastman!

As I understand it, the uploadSession is created:
![image](https://user-images.githubusercontent.com/16662190/115198341-395f2a80-a0f2-11eb-970c-ee2ff300c8fc.png)

_httpClient is also successfully instantiated through DI:
![image](https://user-images.githubusercontent.com/16662190/115198220-1c2a5c00-a0f2-11eb-8653-ea93030e03fe.png)

And I can confirm that all the parameters (storeDriveId, uidFolderId, fileName and uploadProperties) have values. Also memoryStream and maxSliceSize have values.
![image](https://user-images.githubusercontent.com/16662190/115199097-1c772700-a0f3-11eb-9522-f5d2edc91459.png)

The thing that I do notice know is that uploadUrl is empty. If that's the problem, could you please help me understand how to get its value?

**Comment by andrueastman** [comment]
The `uploadUrl` is where the stream/file will be uploaded to and if it is null, it will definitely be an issue. 
I would first suggest first confirming if the values for `storeDriveId` and `uidFolderId` are valid and that you can get its properties by making a call like this and viewing its properties.

```
var driveItem = await graphClient.Drives[storeDriveId].Items[uidFolderId]
	.Request()
	.GetAsync();
```

**Comment by ivancavlek** [comment]
Ok, I found my error 😛.

I was getting the storeDriveId from SharePoint site A and uidFolderId from SharePoint site B. The mix obviously doesn't work (sarc)...

But, when I corrected this error, I still received _Operation is not supported on this platform._ error, although UploadUrl was present. I needed to add the fourth parameter to ```LargeFileUploadTask```, the ```GraphServiceClient``` object, _httpClient in my case. Then it worked like a charm!

Sorry for any inconvenience that I may have caused and thank you for your time and assistance, @andrueastman 😊!
